ruby/example_code/iam/iam-ruby-example-manage-policies.rb

```diff
@@ -1,88 +1,174 @@
-# snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
-# snippet-sourceauthor:[Doug-AWS]
-# snippet-sourcedescription:[Creates a policy, gets its ARN, attaches the policy to a role, lists the policies attached to a role, removes the policy from a role.]
-# snippet-keyword:[AWS Identity and Access Management]
-# snippet-keyword:[attach_role_policy method]
-# snippet-keyword:[create_policy method]
-# snippet-keyword:[detach_role_policy method]
-# snippet-keyword:[get_policy method]
-# snippet-keyword:[list_attached_role_policies method]
-# snippet-keyword:[Ruby]
-# snippet-sourcesyntax:[ruby]
-# snippet-service:[iam]
-# snippet-keyword:[Code Sample]
-# snippet-sourcetype:[full-example]
-# snippet-sourcedate:[2018-03-16]
-# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+# The following code example shows how to:
+# 1. Create a policy in AWS Identity and Access Management (IAM).
+# 2. Attach the policy to a role.
+# 3. List the policies that are attached to the role.
+# 4. Detach the policy from the role.
+
+require 'aws-sdk-iam'
+
+# Creates a policy in AWS Identity and Access Management (IAM).
 #
-# This file is licensed under the Apache License, Version 2.0 (the "License").
-# You may not use this file except in compliance with the License. A copy of the
-# License is located at
+# @param iam_client [Aws::IAM::Client] An initialized IAM client.

```

Comment:
What does "initialized" mean? Is it necessary?

Reply:
"Initialized" in the sense that you must call the "new" method; AWS access credentials must be properly inferred or supplied, the desired AWS Region must be inferred or supplied, etc.